nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/textract/StartTextractJob.java

```diff
@@ -0,0 +1,442 @@
+        .expressionLanguageSupported(FLOWFILE_ATTRIBUTES)
+        .build();
+    static final PropertyDescriptor CLIENT_REQUEST_TOKEN = new PropertyDescriptor.Builder()
+        .name("Client Request Token")
+        .displayName("Client Request Token")
+        .description("A value that will be sent to AWS in order to uniquely identify the request. This prevents accidental job duplication, which could lead to unexpected expenses. For example, if " +
+            "NiFi is restarted and the same FlowFile is sent a second time, this will result in AWS Textract being smart enough not to process the data again. However, if the Processor settings " +
+            "change, this could result in a processing failure. In such a case, this value may need to be changed.")
+        .required(true)
+        .addValidator(NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(FLOWFILE_ATTRIBUTES)
+        .defaultValue("${uuid}")
+        .build();
+    static final PropertyDescriptor JOB_TAG = new PropertyDescriptor.Builder()
+        .name("Job Tag")
+        .displayName("Job Tag")
+        .description("The Job Tag to include in the request to AWS. This can be used to identify jobs in the completion statuses that are published to Amazon SNS.")
+        .required(false)
+        .addValidator(NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(FLOWFILE_ATTRIBUTES)
+        .build();
+    static final PropertyDescriptor KMS_KEY_ID = new PropertyDescriptor.Builder()
+        .name("KMS Key ID")
+        .displayName("KMS Key ID")
+        .description("If specified, the key whose ID is provided will be used to encrypt the results before writing the results to S3")
+        .required(false)
+        .addValidator(NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(FLOWFILE_ATTRIBUTES)
+        .build();
+    static final PropertyDescriptor SNS_TOPIC_ARN = new PropertyDescriptor.Builder()
+        .name("SNS Topic ARN")
+        .displayName("SNS Topic ARN")
+        .description("When Textract completes processing of the document, it will place a notification onto the topic with the specified ARN")
+        .required(true)
+        .addValidator(NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(FLOWFILE_ATTRIBUTES)
+        .build();
+    static final PropertyDescriptor ROLE_ARN = new PropertyDescriptor.Builder()
+        .name("Role ARN")
+        .displayName("Role ARN")
+        .description("The ARN of the Amazon Role that Textract is to use in order to publish results to the configured SNS Topic")
+        .required(true)
+        .addValidator(NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(FLOWFILE_ATTRIBUTES)
+        .build();
+    static final PropertyDescriptor OUTPUT_S3_BUCKET = new PropertyDescriptor.Builder()
+        .name("Output S3 Bucket")
+        .displayName("Output S3 Bucket")
+        .description("The name of the S3 Bucket that the Textract output should be written to")
+        .required(true)
```

Comment:
I'm wondering that couldn't we make this property optional? When I tested I've seen that when Textract finished its job it saved the results to s3, but FetchTextractResults also wrote the result into the FlowFile's content. What happens if I don't want this duplication?